solution

katya-sn opened the pull request:
https://katya-sn.github.io/react_phone-catalog/

maxim2310 reviewed (changes requested):
Your demo link leads to readme file, plz fix it

katya-sn commented:
https://katya-sn.github.io/react_phone-catalog 
I have no idea why at localhost project works correctly, but demo isn`t. Maybe you can help me to fix that?

l4st1m0za reviewed (changes requested):
1) Cheapest works in the wrong direction

2) These links should work
<img width="277" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/67686312/4f465404-1835-490e-84b1-7b8928d27ffb">

3) This seems wrong 
<img width="1426" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/67686312/f64f34c9-ada9-4216-8cc7-8af952187e11">

katya-sn commented:
1. fixed sorting by 'cheapest'
2. I had some errors with available colors (link didn't work) cause in accessories api color write in a different ways ( in colors-available its 'space gray', but color is 'space-gray'). I fixed it by adding normalizedColor;
3. changed path for img in cart

etojeDenys reviewed (changes requested):
1. remove the query param if the query is empty
<img width="99" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/70220667/88f7ce33-a203-4b37-9ef3-95febea9d351">

2. add the favicon
<img width="95" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/70220667/c158e510-c0c3-41c2-ae1b-5032c38d1fe9">

3. I think it better to show the favorites page immediately instead of going back through all the colors and capacities that the user visit


https://github.com/mate-academy/react_phone-catalog/assets/70220667/87f12aa7-784e-422f-a7ad-4284c0cb91f9

TarasHoliuk commented:
Have you redeployed your solution? Because all previous issues are still reproducible, but I see changes in the code. Please, redeploy it.

Also, I've observed a few more things that can be improved to make your demo even better:

1. This button also should be clickable. Also, it would be nice to hide this `Back to top` if there is no scroll at all.
![image](https://github.com/mate-academy/react_phone-catalog/assets/86923089/3474ba79-3302-49ea-b80a-01a6964035e7)

2. The arrow button in the pagination should just set the next page instead of scrolling the pagination itself:

https://github.com/mate-academy/react_phone-catalog/assets/86923089/99ed6e43-b776-4bee-984e-ab34b399da85

3. The footer should always be at the bottom of the page (i.e. set min width for page content, etc.):
![image](https://github.com/mate-academy/react_phone-catalog/assets/86923089/d28e7157-d8de-4716-a660-00be85e2100a)

4. When the cart is empty there is a big empty space in the center. Let's write some text message in that place like `There are no items in the cart yet.`, etc.
![image](https://github.com/mate-academy/react_phone-catalog/assets/86923089/941e85ad-2108-457b-89d5-cc4402cf73bd)

5. All external links should be open in the new tab:
![image](https://github.com/mate-academy/react_phone-catalog/assets/86923089/ae5ea8fd-a4c3-45e6-8a5f-57cffc954eab)

TarasHoliuk reviewed (changes requested):
You did an amazing job! 🔥 Please, redeploy your solution. Also, added a few more small comments 🙂

VitaliyBondarenko1982 reviewed (changes requested):
Good job!
Almost done. Found a couple issues.
- check all pages on different width of screen. for example product details page on width approximately 660px has such view
![Screenshot 2024-05-08 at 22 28 59](https://github.com/mate-academy/react_phone-catalog/assets/33029864/bd765ad9-0b1f-43ae-8bfb-4359006d5a3a)
- some colors from api not exists in css. suggest to make some colors map (object), where key will be color from api and value will be color hash
![Screenshot 2024-05-08 at 22 29 27](https://github.com/mate-academy/react_phone-catalog/assets/33029864/9c82e82c-c61e-49ba-885a-7b499b415a97)
![Screenshot 2024-05-08 at 22 30 16](https://github.com/mate-academy/react_phone-catalog/assets/33029864/40da3f01-fdf8-43a3-91ca-eb9cbee8863d)

GUSILLUS commented:
GJ!

1.Fix category overflow hover
<img width="683" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/97075591/d229cffd-9e09-46a5-af15-c3c15a8e3a5f">

2. I need an ability to remove item from cart 
<img width="330" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/97075591/2c6147ab-22d0-42db-9750-e3277274f7ec">

3. Fix modal ui
<img width="330" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/97075591/d6811a53-712d-4783-828d-47eaccc60e34">

GUSILLUS reviewed (changes requested):
gj

katya-sn commented:
<img width="893" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/132989084/3165de8f-64be-45b1-8c46-24cf8a986d69">

katya-sn commented:
1. I changed hover effect for category (now its transform scale to 0.95)
2. look at previous comment
3. please give me a more detailed comment about notification because I don't understand what the point is

DarkMistyRoom reviewed (changes requested):
Nice job!
- We have 124 models but it looks like there are only 4 pages
<img width="1206" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/98764641/b701ddcb-0f96-4be4-88b5-219449ebd803">

- add some left indent here
<img width="323" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/98764641/376fad41-b325-439e-ae91-5cc0467bb3f1">

- (optional) Would be perfect to make the smooth slider
<img width="1180" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/98764641/37ec458c-2999-4083-81f9-a571096e7bd8">

- It should be possible to remove the item from the cart after the second click on this button, so we can add and remove the item without entering the cart 
<img width="256" alt="image" src="https://github.com/mate-academy/react_phone-catalog/assets/98764641/7321570b-2782-44ad-ba41-807a962ec245">

IvanFesenko reviewed (approved):
Good job. Found a small issue - fix it before move project to portfolio
- Footer logo leads to `page not found`
https://github.com/mate-academy/react_phone-catalog/assets/61904995/08051a42-2ad8-4ff8-90e1-42e402e903b3
